src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -206,18 +206,33 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 {
                     assert(ival == -1);
 
-                    auto emitSwCase = [&](int8_t i) {
-                        emit->emitIns_SIMD_R_R_R_I(ins, simdSize, targetReg, op1Reg, op2Reg, i);
-                    };
+                    auto emitSwCase = [&](int8_t i) { genHWIntrinsic_R_R_RM_I(node, ins, i); };
 
                     if (op3->IsCnsIntOrI())
                     {
                         ssize_t ival = op3->AsIntCon()->IconValue();
                         assert((ival >= 0) && (ival <= 255));
+
+                        if ((intrinsicId == NI_SSE41_Insert) && (baseType == TYP_FLOAT))
+                        {
+                            // Bits 6 and 7 impact the index that is selected from op2
+                            // when op2 is already in register. However, our API exposes
+                            // op2 as a scalar and so bits 6 and 7 must be set to 0.
+                            ival &= 0x3F;
```

Comment:
Added some additional tests to cover this as well.

@eerhardt, we may want to consider adding an overload that is `Vector128<float> Insert(Vector128<float> value, Vector128<float> data, byte index)`. This would allow us to take advantage of the two upper bits (which effectively allow you to extract a given index from `data` and insert it to another given index in `value`, where the two indices do not need to be the same)

Reply:
looks like https://github.com/dotnet/coreclr/issues/18143 tracks this